src/component/reducer/actions/SpectrumsActions.ts

```diff
@@ -79,6 +79,22 @@ function handleChangePeaksMarkersVisibility(draft: Draft<State>, data) {
 function handleChangeActiveSpectrum(draft: Draft<State>, activeSpectrum) {
   let refreshDomain = false;
 
+  // Load specturm range state
+  const range = draft.view.ranges.find(
+    (r) => r.spectrumID === activeSpectrum.id,
+  );
+  if (range) {
+    draft.toolOptions.data = {
```

Comment:
@lpatiny 

is `toolOptions.data` still used anywhere?

It seems like it can be removed now

Reply:
> is `toolOptions.data` still used anywhere and if yes what is its purpose?

I refactored just range tool, `toolOptions.data` still used for other tools states